packages/pyright-internal/src/tests/samples/dataclass26.py

```diff
@@ -0,0 +1,54 @@
+# This sample tests assignment of dataclass fields that use
+# the coverter parameter described in PEP 712.
+
+from dataclasses import dataclass, field
+from typing import Generic, TypeVar
+
+
+def converter_simple(s: str) -> int:
+    return int(s)
+
+
+@dataclass
+class Foo:
+    # This should generate an error because "converter" is not an official property yet.
+    field0: int = field(converter=converter_simple)
+
+foo = Foo("1")
+reveal_type(foo.field0, expected_text="int")
+foo.field0 = "2"
+
+# This should generate an error because only strs can be assigned to field0.
+foo.field0 = 2
+
+
+_T = TypeVar("_T", bound=float)
+
+@dataclass
+class GenericFoo(Generic[_T]):
```

Comment:
I'm thinking that we should not support generic converters for now. Your current code happens to work for this simple case, but it won't work for anything slightly more complicated like a converter that accepts or generates `list[_T]`.